### DISCUSSION TITLE
examples/rules.mk:96: *** multiple target patterns.  Stop. in MCUXpresso IDE on Windows 10 Pro

### DISCUSSION THREAD

**Discussion by focus-martin** [hathach/tinyusb#1555] [Q&A] [answered]
Hi everyone!

I am trying to compile the library to port in the NXP LPC55s28 board. As I want to work in MCUXpresso IDE, I want to include TinyUSB as a submodule of the project inside a folder called `lib`.

Following [this tutorial](https://mcuoneclipse.com/2017/07/22/tutorial-makefile-projects-with-eclipse/), I created an empty Makefile project. Then I created a folder called `lib` and inside `lib` I added TinyUSB as a submodule.

 After that, I replicate the `src` folder and the makefile from one of the examples, at the level of  `lib` folder. Doing so, I had to change the relative paths that are in the Makefile.

From

```
include ../../../tools/top.mk
include ../../make.mk

include ../../rules.mk
```
 to

```
include lib/tinyusb/tools/top.mk
include lib/tinyusb/make.mk

include lib/tinyusb/rules.mk
```

Then, as I am working on Windows, I had to set `SHEEL`  variable to `cmd.exe` because I had problems with the **line 19** in file `top.mk`. 

Finally, I decided to add the definition of `BOARD` at the top of Makefile: `BOARD := lcpxpresso55s28`. And I executed the following command to build: `make -r -j8 all`.

At this point, the result is:

> lib/tinyusb/examples/rules.mk:91: *** multiple target patterns.  Stop.
"make -r -j8 all" terminated with exit code 2. Build might be incomplete.

Line 91 of` rules.mk` is: `$(OBJ): | $(OBJ_DIRS)`

**Does anyone have a suggestion to sort this problem???**

Thanks in advance!!

**Comment by focus-martin** [comment, marked as answer]
Solved!!!

**Reply by hathach** [reply]
would you mind telling how this is solved, it could help other windows users. I don't use windows to build so I could easily miss a thing or two.

**Reply by focus-martin** [reply]
Yes, sure. Sorry I did not explain it before.

The problem was in line #10 of the Makefile:

`SRC_C += $(addprefix $(CURRENT_PATH)/, $(EXAMPLE_SOURCE))`

I had to delete the `addprefix` function, so finally the result was:

`SRC_C += $(EXAMPLE_SOURCE)`

The prefix `CURRENT_PATH` was the problem.
